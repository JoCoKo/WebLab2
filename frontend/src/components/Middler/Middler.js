import React, {useState} from 'react';
import './Middler.css';

const Middler = ({addCity}) => {
    const [text, setText] = useState('');
    const handleChange = (event) => {
        setText(event.target.value)
    };
    const click = (event) => {
        event.preventDefault();
        addCity(text);
        setText('');
    };
    return (
        <div className="Middler">
            <span>Избранное</span>
            <form>
                <input placeholder='Добавить новый город' value={text} onChange={handleChange}/>
                <button onClick={click}>┿</button>
            </form>
        </div>
    );
};

export default Middler;