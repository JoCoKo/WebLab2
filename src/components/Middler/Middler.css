.Middler {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 20px;
    margin-top: 20px;
}

.Middler span {
    font-size: 31px;
}

.Middler input {
    margin-right: 17px;
    height: 40px;
    width: 350px;
    background-color: rgb(241,241,241);
    border-width: 0;
    outline:none;
    padding-left: 10px;
}