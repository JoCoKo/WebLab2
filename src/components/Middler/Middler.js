import React from 'react';
import './Middler.css';

const Middler = () => {
    return (
        <div className="Middler">
            <span>Избранное</span>
            <form>
                <input placeholder='Добавить новый город'/>
                <button>X</button>
            </form>
        </div>
    );
};

export default Middler;