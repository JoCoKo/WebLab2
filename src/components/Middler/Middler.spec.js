import React from 'react';
import Middler from "./Middler"
import {shallowToJson} from "enzyme-to-json";
import {shallow} from "enzyme";

describe('Middler', () => {
  const props = {
    addCity: () => {}
  };
  const middler = shallow(<Middler {...props} />);
  test('отображается корректно', () => {
    expect(shallowToJson(middler)).toMatchSnapshot();
  });
});