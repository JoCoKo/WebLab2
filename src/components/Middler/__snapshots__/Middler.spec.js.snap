// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middler отображается корректно 1`] = `
<div
  className="Middler"
>
  <span>
    Избранное
  </span>
  <form>
    <input
      onChange={[Function]}
      placeholder="Добавить новый город"
      value=""
    />
    <button
      onClick={[Function]}
    >
      ┿
    </button>
  </form>
</div>
`;
